jiant/models.py

```diff
@@ -1154,6 +1176,63 @@ def _lm_forward(self, batch, task, predict):
             pass
         return out
 
+    def _masked_lm_forward(self, batch, task, predict):
+        mlm_probability = 0.15
+        out = {}
+        sent_encoder = self.sent_encoder
+        tokenizer = self.sent_encoder._text_field_embedder.tokenizer
+        # mask_idx = self.sent_encoder._text_field_embedder._mask_id #
+        mask_idx = tokenizer.convert_tokens_to_ids("<mask>")
```

Comment:
better use _mask_id and _pad_id from the self.sent_encoder._text_field_embedder, not every transfromer uses  <mask> as the mask token.

Reply:
in addition, I don't think this is using the right index.
when these indexes get into the transformer, they will be subtracted by 2. the transformer will see them as something other than <mask>.
I'm also not sure about pad_idx. Can you try printing out padding_mask, and see if it is fully zero?

Reply:
I guess padding_mask should be correct, since the targs uses "something_tag" as its namespace. But mask_idx probably isn't.

Reply:
To verify, you can try directly evaluate pretrained model, with no finetuning at all. If it is correctly set to be mask_idx, you will see a fairly low ppl.

Reply:
@HaokunLiu I'll take your suggestion about using the mask_id and _pad_id from the embedder. We fixed the vocab bug by adding a label namespace of _tags to MaskedLanguageModelingTask. And good point about mask_id, i'll check for that. I guess I can add everything (except padding idx) in the inputs by 2 before passing it into the transformer